### PR TITLE
Adjust -i behavior for ln, cp & mv 

### PR DESCRIPTION
**Pull request by sylvestre**
Mentioned in issue #4627
touch a b && echo "n"|./target/debug/coreutils ln -i a b; echo $?

was 0, it is now 1
See: 
https://github.com/coreutils/coreutils/commit/7a69df88999bedd8e9fccf9f3dfa9ac6907fab66

(no problem reading the code here as there isn't much)

### CODE REVIEW

**Review by cakebaker — APPROVED** [review]
While reviewing I noticed that we use "overwrite" in the prompt of `ln` instead of "replace" and fixed it.

File: src/uu/mv/src/mv.rs [hunk 1]
```diff
@@ -420,7 +420,7 @@ fn rename(
             OverwriteMode::NoClobber => return Ok(()),
             OverwriteMode::Interactive => {
                 if !prompt_yes!("overwrite {}?", to.quote()) {
-                    return Ok(());
+                    return Err(io::Error::new(io::ErrorKind::Other, ""));
```
Comment (tertsdiepraam) [inline src/uu/mv/src/mv.rs:423]: Can we do something else than constructing an `io::Error`? This is going to give a weird error message isn't it? For a PR like this, I guess it's OK, though.

**Review by jfinkels — APPROVED** [review]
makes sense

File: src/uu/mv/src/mv.rs (hunk 1 above)
Comment (sylvestre) [reply src/uu/mv/src/mv.rs:423]: well, there isn't any error message :)
```
$ touch a b && echo "n"|./target/debug/coreutils mv -i a b; echo $?
mv: overwrite 'b'? 
1
```

**Pull request merged**